lib/event_tracker.rb

```diff
@@ -0,0 +1,55 @@
+require './event_tracker_haystack.rb'
+
+divider
+
+
+puts "\n\n"
+puts "Requesting Koeln wiki from http://webmontag.de/location/koeln".green
+
+# requesting xhtmlbody file from keoln index.
+begin
+  webmontag_wiki_site = Nokogiri::HTML(open('http://webmontag.de/location/koeln/index?do=export_xhtmlbody'))
+  puts "Extracting website content".green
+rescue
+  puts "Cannot connect to website".red
+  abort
+end
+
+# Create archive object that has the whole li of links.
+# archive => Class: Nokogiri::XML::NodeSet
+#http://www.rubydoc.info/github/sparklemotion/nokogiri/Nokogiri/XML/NodeSet
+archive = webmontag_wiki_site.css('.archive li div')
+if (archive.empty?)
+  puts "\".archive\" selector is empty from the website!".red
+  abort
+end
+
+puts "\n\t #{archive.length} records found".green
+
+
+# This array contains each event from wiki website {name, href}.
+archive_clean = Array.new
+archive.each_with_index do |item, index|
+    # extracting {name, href} form each link and put it in one array (array of hashes).
+    archive_clean.push({name: item.children[0].content.to_s, href: item.children[0].attributes['href'].to_s})
+end
+
+# Create Yaml event-blocks, each block is a hash.
+archive_blocked = Array.new
```

Comment:
as far as I understand this, the `archive_blocked` varialbe just holds a list of events, right?  so maybe just calling it `events` would be more appropiate (and calling the the class `Event`)?